src/webui/api/torrentscontroller.cpp

```diff
@@ -855,17 +996,45 @@ void TorrentsController::addAction()
         }
     };
 
+
     bool partialSuccess = false;
-    for (QString url : asConst(urls.split(u'\n')))
+    for (QString url : urls)
     {
         url = url.trimmed();
-        if (!url.isEmpty())
+        if (url.isEmpty())
+            continue;
+
+        BitTorrent::InfoHash infoHash;
+        if (const auto sourceTorrentDescr = BitTorrent::TorrentDescriptor::parse(url))
+            infoHash = sourceTorrentDescr.value().infoHash();
+        else if (const auto cachedInfoHash = m_torrentSourceCache.get(url))
+            infoHash = cachedInfoHash.value();
+
+        if (const auto torrentDescr = m_torrentMetadataCache.get(infoHash))
+        {
+            if (!filePriorities.isEmpty())
+            {
+                const BitTorrent::TorrentInfo &info = torrentDescr.value().info().value();
+                if (filePriorities.size() != info.filesCount())
+                    throw APIError(APIErrorType::BadParams, tr("Length of filePriorities must equal number of files in torrent"));
```

Comment:
I believe BadParams is wrong error type in this case. Maybe Conflict?

Reply:
File priorities are provided as a parameter by the user, so it seemed most appropriate